### PR TITLE
Adding APIs for anthos machines operations

### PR DESCRIPTION
**Pull request by jainsmit**
**What this PR does / why we need it**:
Adding APIs for anthos machines operations

**Which issue(s) this PR fixes** (optional)
Closes # PTX-23757

**Special notes for your reviewer**:

### CODE REVIEW

File: k8s/anthos/machines.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	// ListMachines lists all machines in kubernetes cluster
+	ListMachine(ctx context.Context) (*v1alpha1.MachineList, error)
+	// GetMachine returns a machine for the given name
+	GetMachine(ctx context.Context, name string) (*v1alpha1.Machine, error)
+	// DeleteMachine delete machine for given name
+	DeleteMachine(ctx context.Context, name string) error
+}
+
+// ListMachines lists all machines in kubernetes cluster
+func (c *Client) ListMachine(ctx context.Context) (*v1alpha1.MachineList, error) {
```
Comment (nikolaypopov) [inline k8s/anthos/machines.go:28]: ListMachines would be a better name for the function and will match your comment on top

Comment (jainsmit) [reply k8s/anthos/machines.go]: Fixed it..

File: k8s/anthos/machines.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	result, err := c.dynamicClient.Resource(machineResource).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	jsonData, err := result.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	machineList := &v1alpha1.MachineList{}
+	if err := json.Unmarshal(jsonData, machineList); err != nil {
```
Comment (kphalgun-px) [inline k8s/anthos/machines.go:42]: I think we can directly Unmarshal the result into `machineList` without converting it into JSON.

```go
result, err := c.dynamicClient.Resource(machineResource).List(ctx, metav1.ListOptions{})
if err != nil {
	return nil, err
}
machineList := &v1alpha1.MachineList{}
if err := json.Unmarshal(result, machineList); err != nil {
	return nil, err
}
```

Comment (jainsmit) [reply k8s/anthos/machines.go:42]: @kphalgun-px  here result will be  [`*unstructured.UnstructuredList` ](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured)  so MarshalJSON() has to be used. How will you parse Unstructured data to json ?

Comment (kphalgun-px) [reply k8s/anthos/machines.go:42]: My bad. Thanks I thought results will be in byte slice format.

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Review by nikolaypopov — APPROVED** [review]

**Pull request merged**